### PR TITLE
Develop movie service with neo4j and graphQL

### PR DESCRIPTION
**Pull request by amerkul**

### CODE REVIEW

File: src/main/java/com/example/neo4j/service/impl/MovieServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    @Override
+    @Transactional
+    public Mono<Void> delete(long id) {
+        return this.repository.deleteById(id);
+    }
+
+    @Override
+    @Transactional
```
Comment (h1alexbel) [inline src/main/java/com/example/neo4j/service/impl/MovieServiceImpl.java]: @amerkul should be transactional on a read-only level for all operations, that don't modify the data
by default, it is write level, which is stricter

File: src/main/java/com/example/neo4j/service/impl/PersonServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    @Override
+    @Transactional
+    public Mono<Void> delete(long id) {
+        return this.repository.deleteById(id);
+    }
+
+    @Override
+    @Transactional
```
Comment (h1alexbel) [inline src/main/java/com/example/neo4j/service/impl/PersonServiceImpl.java]: @amerkul 
the same here

File: src/main/java/com/example/neo4j/service/impl/PersonServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    public Flux<Person> retrieveAll() {
+        return this.repository.findAll();
+    }
+
+    @Override
+    @Transactional
```
Comment (h1alexbel) [inline src/main/java/com/example/neo4j/service/impl/PersonServiceImpl.java]: @amerkul the same here

File: src/main/java/com/example/neo4j/service/impl/MovieServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    public Mono<Long> countMovies() {
+        return this.repository.count();
+    }
+
+    @Override
+    @Transactional
```
Comment (h1alexbel) [inline src/main/java/com/example/neo4j/service/impl/MovieServiceImpl.java]: @amerkul the same here

File: src/main/java/com/example/neo4j/service/impl/MovieServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    public Mono<Movie> retrieveByTitle(String title) {
+        return this.repository.findByTitle(title);
+    }
+
+    @Override
+    @Transactional
```
Comment (h1alexbel) [inline src/main/java/com/example/neo4j/service/impl/MovieServiceImpl.java]: @amerkul the same here

File: src/main/java/com/example/neo4j/service/impl/MovieServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    public Flux<Movie> retrieveAll() {
+        return this.repository.findAll();
+    }
+
+    @Override
+    @Transactional
```
Comment (h1alexbel) [inline src/main/java/com/example/neo4j/service/impl/MovieServiceImpl.java]: @amerkul the same here

File: src/test/java/com/example/neo4j/data/PersonData.java [hunk 7]
```diff
@@ -0,0 +1,15 @@
+package com.example.neo4j.data;
+
+import com.example.neo4j.domain.Person;
+import reactor.core.publisher.Flux;
+
+public class PersonData {
+
+    public static Flux<Person> getPeople() {
```
Comment (h1alexbel) [inline src/test/java/com/example/neo4j/data/PersonData.java]: @amerkul test data should be visible in tests

**Review by h1alexbel — CHANGES_REQUESTED** [review]

File: src/test/java/com/example/neo4j/data/PersonData.java (hunk 7 above)
Comment (amerkul) [reply src/test/java/com/example/neo4j/data/PersonData.java]: What exactly do you mean?

Comment (h1alexbel) [reply src/test/java/com/example/neo4j/data/PersonData.java]: @amerkul instead of calling `getPeople()` in test, create your people inside the test

Comment (amerkul) [reply src/test/java/com/example/neo4j/data/PersonData.java]: But I have to use this data into different test classes. I'm planning to create tests on service classes.

**Review by h1alexbel — APPROVED** [review]
lgtm

**Pull request merged**